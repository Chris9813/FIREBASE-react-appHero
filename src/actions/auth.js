import Swal from "sweetalert2"
import {firebase, googleAuthProvider} from "../firebase/firebase-config"
import { types } from "../types/types"


export const startLoginEmailPaswword = (email, password) => {
    return (dispatch) => {
        return firebase.auth().signInWithEmailAndPassword(email, password)
        .then(({user}) => {
            console.log(user);
            dispatch(login(user.uid, user.displayName))
        }).catch(e => {
            console.log(e);
            Swal.fire("Error", e.message, "error")
        })
        
    }
}


export const startGoogleLogin = () =>{
    return (dispatch) =>{
        firebase.auth().signInWithPopup( googleAuthProvider )
            .then( userCred =>  {
                console.log(userCred);
            })
    }
}


export const login = (uid, displayname) => ({
    type: types.login,
    payload: {
        uid,
        displayname,
    }
})

export const startLogout = () => {
    return async(dispatch) => {
        await firebase.auth().signOut()
        dispatch(logout())
    }
}

export const logout = () => ({
    type: types.logout
})


