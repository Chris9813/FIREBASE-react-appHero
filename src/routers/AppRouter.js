import React, { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
    import {
        BrowserRouter as Router,
        Switch
    } from "react-router-dom";
import { LoginScreen } from '../components/login/LoginScreen';
import { DeshboardRoutes } from "./DeshboardRoutes";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";
import {firebase} from '../firebase/firebase-config'
import { login } from "../actions/auth";

export const AppRouter = () => {

    const dispatch = useDispatch()

    const [checking, setchecking] = useState(true);
    const [isLoggedIn, setisLoggedIn] = useState(false)


    useEffect(() => {
        firebase.auth().onAuthStateChanged(async(user) =>{
            if(user?.uid) {
                dispatch(login(user.uid, user.displayName))
                setisLoggedIn(true)
            } else {
                setisLoggedIn(false)
            }
            setchecking(false)
        })
    }, [dispatch, setchecking, setisLoggedIn])


if(checking) {
    return(
        <h1>Wait...</h1>
    )
}

    return (
        <div>
        <Router>
    <div>
        <div>
        <Switch>
        <PublicRoute
        exact path = "/login" component = {LoginScreen}
        isAuthenticated = {isLoggedIn}
        />

        <PrivateRoute
        path="/" 
        component = {DeshboardRoutes} 
        isAuthenticated = {isLoggedIn}
        />

        </Switch>
        </div>
        
    </div>
    </Router>
        </div>
        
    )
}
