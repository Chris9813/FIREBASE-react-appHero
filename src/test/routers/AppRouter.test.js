import { mount, shallow } from "enzyme"
import { AuthContext } from "../../auth/AuthContext"
import { AppRouter } from "../../routers/AppRouter"


describe('Prueas en el componente <AppRouter />', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('debe mostrar el login si no esta autenticado ', () => {
        const wrapper = mount(
            <AuthContext.Provider value = {contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )
        console.log(wrapper.html());
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('debe mostra el componente marvel si esta autenticado ', () => {
        
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: "CHRIS"
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value = {contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )
        expect(wrapper.find(".navbar").exists()).toBe(true)
    })
    

})
