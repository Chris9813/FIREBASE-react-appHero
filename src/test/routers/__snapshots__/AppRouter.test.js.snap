// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueas en el componente <AppRouter /> debe mostrar el login si no esta autenticado  1`] = `
<div>
  <div>
    <div>
      <div>
        <div
          className="container mt-5"
        >
          <h1>
            Login Screen
          </h1>
          <hr />
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
